classes/event/grading_response_failed.php

```diff
@@ -0,0 +1,66 @@
+<?php
+// This file is part of the QuestionPy Moodle plugin - https://questionpy.org
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace qtype_questionpy\event;
+
+use moodle_exception;
+use moodle_url;
+
+/**
+ * Grading attempt failed event.
+ *
+ * @package    qtype_questionpy
+ * @author     Jan Britz
+ * @copyright  2024 TU Berlin, innoCampus {@link https://www.questionpy.org}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class grading_response_failed extends \core\event\base {
+    /**
+     * Initialise event parameters.
+     */
+    protected function init() {
+        $this->data['crud'] = 'c';
+        $this->data['edulevel'] = self::LEVEL_TEACHING;
+    }
+
+    /**
+     * Returns localised event name.
+     *
+     * @return string
+     * @throws moodle_exception
+     */
+    public static function get_name() {
+        return get_string('event_grading_response_failed', 'qtype_questionpy');
+    }
+
+    /**
+     * Returns non-localised event description with id's for admin use only.
+     *
+     * @return string
+     */
+    public function get_description() {
+        return $this->other['description'];
```

Comment:
Description sollte hier zusammengebaut werden. Nur die Werte/IDs, die in den Text eingebaut werden, sollten in `$this->other` gespeichert werden. Erstens spart man dadurch Speicherplatz in der DB, zweitens kann man die Formulierung dann noch nachträglich ändern.